src/request.js

```diff
@@ -65,11 +61,11 @@ export default class Request {
 			signal = input[INTERNALS].signal;
 		}
 
-		if (init.signal != null) {
-			if (!isAbortSignal(init.signal)) {
+		if (init.signal !== undefined) {
+			if (init.signal !== null && !isAbortSignal(init.signal)) {
 				throw new TypeError('Provided signal must be an AbortSignal object');
 			}
-			signal = init.signal;
+			signal = init.signal === null ? undefined : init.signal;
 		}
```

Comment:
[whatwg](https://fetch.spec.whatwg.org/#request-class)(30) says that signal should be followed only if it's not `null`. I believe that when we provide `null`, we should no longer follow "parent's" signal. Or did I misinterpret it?

Reply:
Eek, this is one of the areas the Web IDL spec is pretty unclear on. I think your handling is correct in this case, as I believe the spec writers wanted to distinguish null vs. undefined for exactly this purpose.

Reply:
Since the current Request has been resolved, we must remove it from signal's event listeners so it can be garbage collected. Unless I missed something, Response cannot be aborted. I did, however, forgot to propagate signal when redirecting.